library/std/src/env.rs

```diff
@@ -353,15 +356,26 @@ impl Error for VarError {
 /// use std::env;
 ///
 /// let key = "KEY";
-/// env::set_var(key, "VALUE");
+/// unsafe {
+///     env::set_var(key, "VALUE");
+/// }
 /// assert_eq!(env::var(key), Ok("VALUE".to_string()));
 /// ```
+#[cfg(not(bootstrap))]
+#[rustc_deprecated_safe_2024]
```

Comment:
```suggestion
#[cfg_attr(not(bootstrap), rustc_deprecated_safe_2024)]
```
The `unsafe` block in the body can be used unconditionally.

Reply:
The difference between the `#[cfg(bootstrap)]` and `#[cfg(not(bootstrap))]` is mostly whether the function itself is `unsafe` – I'm not aware of a method to conditionally mark a function as unsafe.